provider/views.py

```diff
@@ -270,50 +369,31 @@ def handle(self, request, post_data=None):
 
         authorization_form = self.get_authorization_form(request, client, post_data, data)
 
-        if data.get('response_type', None):
-            if not (authorization_form.is_bound and authorization_form.is_valid()):
-                return self.render_to_response({
-                    'client': client,
-                    'form': authorization_form,
-                    'oauth_data': data,
-                })
-
-            if not authorization_form.cleaned_data['authorize']:
-                # User hit cancel.
-                response_data = {
-                    "error": "access_denied",
-                    "error_description": "The user denied the scope request.",
-                    "state": data.get('state'),
-                }
-
-                url = "{0}?{1}".format(
-                    client.redirect_uri, urllib.urlencode(response_data))
-
-                return HttpResponseRedirect(url)
-
-            if data.get('response_type') == 'token':
-                # uses request.user as the urls.py already required login_required
-                lookup_kwargs = {
-                    "user": request.user,
-                    "client": client,
-                }
-
-                if data.get('scope'):
-                    lookup_kwargs['scope'] = data.get('scope')
-
-                atm = None
-                if constants.SINGLE_ACCESS_TOKEN:
-                    try:
-                        atm = AccessTokenModel.objects.get(
-                            expires__gt=now(), **lookup_kwargs)
-                    except AccessTokenModel.DoesNotExist:
-                        pass  # Fall through to non-single-access token case
-
-                if not atm:
-                    atm = AccessTokenModel.objects.create(**lookup_kwargs)
-
-                at = AccessToken()
-                return at.access_token_response(atm, data)
+        if not (authorization_form.is_bound and authorization_form.is_valid()):
+            return self.render_to_response({
+                'client': client,
+                'form': authorization_form,
+                'oauth_data': data,
+            })
+
+        if not authorization_form.cleaned_data['authorize']:
+            # User hit cancel.
+            response_data = {
+                "error": "access_denied",
+                "error_description": "The user denied the scope request.",
+                "state": data.get('state'),
+            }
+
+            url = "{0}?{1}".format(
+                client.redirect_uri, urllib.urlencode(response_data))
+
+            return HttpResponseRedirect(url)
+
+        if 'token' in data['response_type']:
```

Comment:
Ah.  Thanks for the explanation.  To be safer, consider adding a `.split(" ")` at the end.  Or, even better, update `clean_response_type` to return a list of response_types instead of the space-separated string.